### PR TITLE
Integration of docker & sequelize into todo-API

### PR DESCRIPTION
**Pull request by Ljgleeson**

### CODE REVIEW

File: repo/iRepository.ts [hunk 1]
```diff
@@ -1,86 +0,0 @@
...
-            return this.task_list.splice(index, 1)
-        }else{
-            return null
-        }
-    }
-}
-
-//export object to routes so can be used globally
-export const task_Repo = new taskRepository()
-
```
Comment (lawhorkl) [inline repo/iRepository.ts:86]: Do not remove the old repository implementations

File: src/repo/iRepository.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    host: 'localhost',
+    port: 3306,
+    dialect: 'mysql'
+} )
+
+dbConnection.authenticate().then(() => {
+    console.log("Database connected")
+}).catch( () => {
+    console.log("Database couldnt connect")
+})
```
Comment (lawhorkl) [inline src/repo/iRepository.ts:14]: I dont think this is using the hapi plugin?

File: src/repo/iRepository.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+//implementation of generic repository interface 
+class taskRepository implements IRepo<Task>  {
+    
+    async create(T: Task){
+        //Used ? for values as couldnt directly assign task values and then replaced them beneath
+        const [results, metadata] = await dbConnection.query(
+            'INSERT INTO tasks VALUES ( ?, ?, ?, ?, ?)', {
+                replacements: [T.id, T.createdAt, T.title, T.dueDate, T.completed]
+            } )
```
Comment (lawhorkl) [inline src/repo/iRepository.ts:45]: Replacements, A+.

My question now is, WHY do you use replacements? Does the syntax just look cooler?

File: src/repo/iRepository.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            return results
+        }else if (val == 'createdAt'){
+            const [results, metadata] = await dbConnection.query('SELECT * FROM tasks ORDER BY createdAt ASC')
+            return results
+        }else if (val == 'dueDate'){
+            const [results, metadata] = await dbConnection.query('SELECT * FROM tasks ORDER BY dueDate ASC')
+            return results
+        }else{
+            return null
+        }
```
Comment (lawhorkl) [inline src/repo/iRepository.ts:93]: Couldnt you just parse the column to sort on rather than a switch statement?

**Review by lawhorkl — APPROVED** [review]
Great work.

File: src/repo/iRepository.ts (hunk 3 above)
Comment (Ljgleeson) [reply src/repo/iRepository.ts:45]: Replacements sanitize the query before using it to make sure it hasn't been modified. This prevents sql injections from occurring by removing any special characters that could have been added to the original query.

Comment (lawhorkl) [reply src/repo/iRepository.ts:45]: How do you sanitize inputs?

Comment (Ljgleeson) [reply src/repo/iRepository.ts:45]: By validating the user input is the correct data type. This could be done using the Joi plugin

**Pull request merged**